src/chains/ethereum/ethereum/src/data-managers/block-manager.ts

```diff
@@ -22,6 +22,8 @@ const NOTFOUND = 404;
 
 const EMPTY_BUFFER = Buffer.from([]);
 
+type RawOrBlock<T> = T extends true ? Buffer : Block;
```

Comment:
In order to understand what `T` means, I need to look at the conditional type.

Could we rename it to something like `IsRaw` ?

Additionally - we should constraint it's type to boolean.

```suggestion
type RawOrBlock<IsRaw extends boolean> = IsRaw extends true ? Buffer : Block;
```

Reply:
Great call, thank you!